### ISSUE TITLE
Missing _t and _v in vector traits

### ISSUE THREAD

**Issue by AnastasiaStulova** [KhronosGroup/libclcxx#63]
Some vector traits seem to be missing `_t` and `_v` versions, for example `vector_size_t` is missing.

**Comment by AnastasiaStulova** [comment]
duplicates https://github.com/KhronosGroup/libclcxx/issues/60

**Comment by 24spoorthi** [comment]
Hey, while trying to work on this issue, when I tried to clone the repo and compile the code, I encountered errors that was related to the opencl_type_traits.
The following was the code I tried to execute, even when there was no use of the header file error was thrown.
```
#include<iostream>
#include <opencl_type_traits>

using namespace std;

int main(){
        cout<<"Hello world";
        return 0;
}
```
The command used to execute was as follows:
```
clang -I include/ new.cpp
```

The error looked something like this:
![Error on compilation](https://user-images.githubusercontent.com/57169173/179555835-0026b45a-25cb-4578-8363-9f30bf0d1782.png)

Could you please help me out with this?

**Comment by AnastasiaStulova** [comment]
Clang compiles `.cpp` files in pure C++ mode, so to compile in C++ for OpenCL you need to use `.clcpp` file extension or add `-cl-std=clc++`.

Examples:
```
clang -I include/ new.clcpp
clang -cl-std=clc++ -I include/ new.cpp
```

I would suggest using `.clcpp` file extensions, however it is only supported from clang 13 onwards.

Btw `iostream` is not yet supported by libclcxx project so you will likely experience issues when including it in your sources.

**Comment by 24spoorthi** [comment]
To get a better understanding of this issue and to confirm my understanding, these ```scalar_type``` returns a native datatype, for example, 
```std::scalar_type<uint8>::type```  will create a scalar type of the vector uint8, which is ```int```.

Similarly, I wanted to confirm, as an example, what does ```scalar_type_t``` accept as a parameter and what does it return? Similarly, for ```scalar_type_v``` .

Are these simply an alias to ```scalar_type```, say for example,
```std::scalar_type_t<uint8_t>::type``` creates a scalar type of the vector uint8 which is  ```int``` or is it something different?

Looking forward to hearing back.

**Comment by AnastasiaStulova** [comment]
> Are these simply an alias to scalar_type, say for example,
> std::scalar_type_t<uint8_t>::type creates a scalar type of the vector uint8 which is int or is it something different?

Exactly. The idea is to provide similar functionality to traits from C++17. For example `remove_pointer` has the following alias:

```
template< class T >
using remove_pointer_t = typename remove_pointer<T>::type;
```

which means we can use these two formats interchangeably:

```
std::is_same_v<int, std::remove_pointer<int*>::type>
std::is_same_v<int, std::remove_pointer_t<int*>>
```

Basically this is just to avoid writing `::type` or `::value` which makes things longer.
